### PR TITLE
Fix accessibility issues with page layouts

### PR DESCRIPTION
**Pull request by adamwoodnz**
See #368 

Several issues with the handbook layouts are fixed, and some of these fixes have also been applied to other layouts (remove search field from main landmark):

- [x] The handbook chapters heading 2 is the first heading on the page. This design needs to be reworked, no questions asked.
- [x] The skip to main content link does not land you in the main content, it lands you in all the items before the main heading 1.
- [x] The main landmark should not wrap navigational elements such as the Chapters heading 2 or the search form above it.

**The visual layout should not change.**

### Testing

Use dev tools to test and check the following:

1. Almost all the templates have updated layouts, with the Search field moving out of `main`. The skip to content link should take you directly to the main content, with no search or chapters in main before the h1.
2. On handbook pages the chapter list aside has been moved after main, so check the headings hierarchy and accessibility tree

### CODE REVIEW

File: source/wp-content/themes/wporg-developer-2023/src/handbook-pagination/view.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Add screen reader text to handbook pagination links.
+ * This is needed because the post-navigation-link block cannot accept rich content via the label attribute.
```
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/src/handbook-pagination/view.js:8]: I tried passing this content through the label attribute. Passing the HTML string in directly as `<span aria-hidden="true">Next </span><span class="screen-reader-text">Next chapter: </span>` worked, but using `wp_kses_post` didn't, I think because it broke the block markup. The rendered block was missing all the passed attributes.

```php
wp_kses_post(__(<span aria-hidden="true">${ visualLabel }</span><span class="screen-reader-text">${ screenReaderText }</span>)
```

File: source/wp-content/themes/wporg-developer-2023/templates/single-handbook-github.html
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/templates/single-handbook-github.html:1]: No longer required now that the pagination has built in logic to determine whether to show GitHub links

File: source/wp-content/themes/wporg-developer-2023/src/article-meta-github/block.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		do_blocks( '<!-- wp:paragraph {"className":"external-link"} --><p class="external-link"><a href="[article_edit_link]">%s</a></p><!-- /wp:paragraph -->' ),
+		sprintf(
+			/* translators: %s: article title */
+			__( 'Improve it on GitHub<span class="screen-reader-text">: %s</span>', 'wporg' ),
+			esc_html( $title )
+		)
+	);
+	$content .= do_blocks( '</div><!-- /wp:group -->' );
+
+	if ( 'blocks-handbook' !== $post_type ) {
```
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/src/article-meta-github/block.php:68]: See https://github.com/WordPress/wporg-developer/issues/345

**Comment by StevenDufresne** [comment]
This PR is hard to follow as there are abstractions/refactoring and format updates. Is there an easier way to look at this? Otherwise, I don't think I can provide a useful code-specific review.

**Comment by adamwoodnz** [comment]
> This PR is hard to follow as there are abstractions/refactoring and format updates. Is there an easier way to look at this? Otherwise, I don't think I can provide a useful code-specific review.

Yeah that's fair, there's a lot going on. I'll split it in 3; the 2 new blocks and the layout changes.

**Comment by adamwoodnz** [comment]
> there are abstractions/refactoring and format updates. Is there an easier way to look at this?

The scope of this has been reduced to only include the layout changes, see updated description 🙏
